packages/forms/src/UIForm/fields/Datalist/Datalist.component.js

```diff
@@ -89,7 +89,7 @@ class Datalist extends Component {
 
 		let titleMapFind = titleMap;
 
-		if (!restricted) {
+		if (!restricted && propsValue) {
```

Comment:
Can you check with an explicit condition rather than with truthy/falsy check ?
And also swap the condition to an early return.

Reply:
`propsValue` is a string, and we don't want to add it if it's empty or undefined.
--> basically it's a falsy check
I'll let that check, but I change the condition to have an early return

Reply:
It's just implicit vs explicit, it tells many things :
- we expect a string (no need to go check the props def) => yes a typed language would be more convenient ...
- it tells you have added this condition strictly for checking empty string and nothing else (null, undefined, whatever). And if the prop type change, it will need some revision to check if condition is handle properly. It's not hidden and will seem to be checked where it's not.

It's just about communicating intention explicitly and avoid useless thinking.
But whatever, yes it's working